cfme/services/catalogs/catalog_item.py

```diff
@@ -152,7 +152,7 @@ def create(self):
                                'select_catalog': catalog.name,
                                'select_dialog': dialog.name,
                                'select_orch_template': self.orch_template,
-                               'select_provider': self.provider,
```

Comment:
Yes but here we need the provider for some catalog items (like ec2 and openstack) and don't need it for other (like vmware and Rhev) . All of these catalog items also need provider to select template.
So I need to pass the provider object to select the template but also need to pass provider_type which will be none for vmware and Rhev and will have some value (same as provider ) for ec2 and openstack.